### PR TITLE
fix flaky test test_rai_insights_add_save_load_save on dataset fetch

### PR DESCRIPTION
**Pull request by imatiach-msft**
## Description

The test test_rai_insights_add_save_load_save is flaky has been erroring including on the build:

https://github.com/microsoft/responsible-ai-toolbox/actions/runs/3087387703/jobs/4992705136

The underlying error was a networking issue:

```
>           raise FileNotFoundError(f"{path} not found.")
E           FileNotFoundError: https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data not found.
```

Adding simple retry logic around it via raiutils retry_function method makes the test more reliable.

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: responsibleai/tests/common_utils.py [hunk 1]
```diff
@@ -189,8 +191,23 @@ def create_housing_data(create_small_dataset=True):
...
+        pass
+
+    def fetch(self):
+        return helpers.load_adult_income_dataset()
+
+
 def create_adult_income_dataset(create_small_dataset=True):
-    dataset = helpers.load_adult_income_dataset()
+    fetcher = FetchDiceAdultCensusIncomeDataset()
+    action_name = "Dataset download"
```
Comment (gaugup) [inline responsibleai/tests/common_utils.py:204]: Can we change the action name to "Adult dataset download"?

File: responsibleai/tests/common_utils.py [hunk 2]
```diff
@@ -189,8 +191,23 @@ def create_housing_data(create_small_dataset=True):
...
+    err_msg = "Failed to download dataset"
```
Comment (gaugup) [inline responsibleai/tests/common_utils.py:205]: Can we change the err_msg to be "Failed to download adult dataset"?

File: responsibleai/tests/common_utils.py (hunk 1 above)
Comment (imatiach-msft) [reply responsibleai/tests/common_utils.py:204]: done!

File: responsibleai/tests/common_utils.py (hunk 2 above)
Comment (imatiach-msft) [reply responsibleai/tests/common_utils.py:205]: done!

**Review by gaugup — APPROVED** [review]

**Pull request merged**